_posts/2020-08-15-Named-Graphs.md

```diff
@@ -0,0 +1,539 @@
+| http://example.com/LL/síl_daimini.trig#Fhind-a5cb5100                 |
+
+Which one of these results is correct?  Both are if the user considers
+that GraphDB creates an equivalence class that combines answers when
+doing inferencing.  The two answers can be 
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+prefix onto: <http://www.ontotext.com/>
+
+select ?a
+from <http://example.com/LL>
+from onto:explicit
+where {
+    ?a irishRel:nomName "Find"
+}
+```
+
+What is `onto:explicit`?  As the GraphDB documentation discussed above states:
+
+> FROM onto:explicit – using only this clause (or with FROM onto:disable-sameAs) produces the same results as when the inferencer is disabled (as with the empty ruleset). This means that the ruleset and the disable-sameAs parameter do not affect the results.
+
+This disables the inferencer and returns what is explicitly stated in
+the IrishGen files.  Is this what the user will always want though?
+That is up to the user at the tie the query is run.
+
+To return to the query above which contains two Named Graphs, if the
+user wished to have all the instances of `irishRel:nomName "Find"`
+from LL and Rawl B502 without the `owl:sameAs` equivalencies
+interfering with the results, the query would be:
+
+```sparql
+prefix onto: <http://www.ontotext.com/>
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a ?g
+from <http://example.com/Rawl_B502>
+from named <http://example.com/LL>
+from onto:explicit
+from named onto:explicit
+where {
+    { ?a irishRel:nomName "Find" }
+    union {
+		graph ?g {
+			?a irishRel:nomName "Find"
+		}
+    }	
+}
+```
+
+Although, it would be easier to do the following:
+
+```sparql
+prefix onto: <http://www.ontotext.com/>
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a ?g
+from named <http://example.com/Rawl_B502>
+from named <http://example.com/LL>
+from named onto:explicit
+where {
+	graph ?g {
+		?a irishRel:nomName "Find"
+	}
+		
+}
+```
+
+The above is easier to write and read.  This returns 44 results, which
+contrasts with the 21 results returned from Stardog.  To bring the
+results into alignment, all one needs to do is remove the
+`onto:explicit` and the above query will return 21 just the same as
+Stardog.
+
+As the reader can see, using OWL 2's `sameAs` has various effects on
+the outcome of queries.  This needs to be kept in mind whenever a
+query is run.  A user must keep in mind the effects of choosing one
+Triplestore over another as each Triplestore makes different choices
+on how to represent the OWL 2 standard and these choices can have a
+large effect on the result sets returned.
+
+# Conclusion
+
+Named Graphs allow IrishGen to partition the various MSS sources into
+their own sub-graphs.  This allows the user to interrogate one MS
+tradition or another, which can be important to certain investigations
+such as when a user is searching broadly for some information; for
```

Comment:
Searching broadly for some information is not an example of when a user might want to "interrogate one MS tradition or another", though. Perhaps rephrase this?